pkg/dispatcher/repository_db.go

```diff
@@ -15,14 +15,17 @@ type Repository struct {
 	logger     log.Logger
 }
 
-func NewRepository(config cfg.Config, logger log.Logger, repo db_repo.Repository) db_repo.Repository {
-	disp := Get()
+func NewRepository(ctx context.Context, config cfg.Config, logger log.Logger, repo db_repo.Repository) (db_repo.Repository, error) {
+	disp, err := ProvideDispatcher(ctx, config, logger)
+	if err != nil {
+		return nil, err
```

Comment:
We should wrap the error here to add some context like "can not get dispatcher" or something like that..

Reply:
done